libraries/rush-lib/src/api/Rush.ts

```diff
@@ -102,7 +102,7 @@ export class Rush {
     options = Rush._normalizeLaunchOptions(options);
 
     Rush._assignRushInvokedFolder();
-    RushXCommandLine._launchRushXInternal(launcherVersion, { ...options });
+    RushXCommandLine.launchRushX(launcherVersion, options);
```

Comment:
See the changes in RushXCommandLine.ts for the technical reasons motivating this change.   It also doesn't seem kosher to call the internal version from here.

Reply:
It's marked internal, not private. The marker is that it is expected to be called from within the package and not externally. These are in the same package.

Reply:
I see.  Shall I leave the code as-is, or add another layer to contain the code added to launchRushX?

Reply:
I'm trying to remember why we did this like this. @octogonz, do you recall?

Reply:
Looking at the code, the sort-of-public entry points are:
```js
export class Rush {
    static launch(launcherVersion: string, arg: ILaunchOptions): void;
    static launchRushPnpm(launcherVersion: string, options: ILaunchOptions): void;
    static launchRushX(launcherVersion: string, options: ILaunchOptions): void;
```

They are not marked as `@internal` but instead have doc comments instructing third parties NOT to use them, because they are only meant to be accessed by the `@microsoft/rush` CLI front end.  

Why didn't we label them as `@internal`? I think the answer is that `Rush.launch()` is ancient and predates API Extractor, and then the other two just continued the same pattern.  We "should" fix that, except that doing so would break compatibility with all previous versions of the `@microsoft/rush` front end.

The situation of `RushXCommandLine` is more mysterious, since this class is not actually exported by `index.d.ts` at all, yet it has both an `@internal` and regular method:

```ts
export class RushXCommandLine {
  public static launchRushX(launcherVersion: string, isManaged: boolean): void {
    RushXCommandLine._launchRushXInternal(launcherVersion, { isManaged });
  }

  /**
   * @internal
   */
  public static _launchRushXInternal(launcherVersion: string, options: ILaunchRushXInternalOptions): void {
```

I forget the details, but my guess is that there is probably one early release of `@microsoft/rush` that did import `RushXCommandLine.launchRushX()` but then we quickly replaced that with `Rush.launchRushX()` to reduce the API surface.  Rush 6 would be a good opportunity to clean up this contract.

Reply:
> I see. Shall I leave the code as-is, or add another layer to contain the code added to launchRushX?

@nebosite my recommendation is revert these changes if you want to get your PR merged quickly. 😄

Reply:
@octogonz To be clear, for this particular line of code,  it is a _repointing_, not a renaming.   Both launchRushx and _launchRushXInternal have the same names and this repointing is necessary to preserve the same functionality as before.

Reply:
That's true here, but in the `RushXCommandLine` class, this API:

```ts
  public static launchRushX(launcherVersion: string, isManaged: boolean): void {
```
...got renamed to this:
```ts
 public static launchRushX(launcherVersion: string, options: ILaunchRushXInternalOptions): void {
```

I'm not necessarily opposed to cleaning this up, but since it's old code and we didn't remember whether there are `@microsoft/rush` releases that access it, it might require some additional review.

Whereas the `RushXCommandLine._launchRushXInternal()` is explicitly marked as `@internal`, so it seems very safe to change that one however you like, including its `ILaunchRushXInternalOptions` interface.

Reply:
(Hold on, I'm going to just look at the Git history.)

Reply:
- The original implementation of `rushx` did not have `RushXCommandLine` at all: 
   https://github.com/microsoft/rushstack/pull/678
- That file was added by some refactoring, but initially without introducing the `@internal` member `_launchRushXInternal()`: 
   https://github.com/microsoft/rushstack/pull/684
- The `@internal` member was added in this later PR:
   https://github.com/microsoft/rushstack/pull/1388

The review comments don't explain why we added an `@internal` member to `RushXCommandLine`, but perhaps it was this same paranoia about whether that was a `@microsoft/rush` version selector contract or not.

So, looking at this code, I think we can conclude that `RushXCommandLine` is entirely internal to `rush-lib`, and therefore we should remove the misleading `@internal` markers from that class, and you're free to refactor it however you like. 🙂

Reply:
OK, I removed the internal markers and refactored the call to _launchRushXInternal to take a single options argument.